sdk/eventhub/event-hubs/README.md

```diff
@@ -1,259 +1,198 @@
+  - This method takes the connection string and entity name to your Event Hub instance. You can get the connection string
+    from the Azure portal
+- [createFromIotHubConnectionString](https://docs.microsoft.com/en-us/javascript/api/%40azure/event-hubs/eventhubclient?view=azure-node-latest#createfromiothubconnectionstring-string--clientoptions-)
+  - This method takes an IotHub Connection string. You can get the connection string from the Azure portal. This is useful for receiving telemetry data
+    of IotHub from the linked EventHub. Most likely the associated connection string will not have send claims. Hence getting HubRuntimeInfo or PartitionRuntimeInfo
+    and receiving events would be the possible operations.
+- [createFromTokenProvider](https://docs.microsoft.com/en-us/javascript/api/%40azure/event-hubs/eventhubclient?view=azure-node-latest#createfromtokenprovider-string--string--tokenprovider--clientoptionsbase-)
+  - This method takes the host name and entity name of your Event Hub instance and your custom Token Provider. The
+    host name is of the format `name-of-event-hub-instance.servicebus.windows.net`.
+- [createFromAADTokenCredentials](https://docs.microsoft.com/en-us/javascript/api/%40azure/event-hubs/eventhubclient?view=azure-node-latest#createfromaadtokencredentials-string--string--applicationtokencredentials---usertokencredentials---devicetokencredentials---msitokencredentials--clientoptionsbase-)
+  - This method takes the host name and entity name of your Event Hub instance and a credentials object that you need
+    to generate using the [@azure/ms-rest-nodeauth](https://www.npmjs.com/package/@azure/ms-rest-nodeauth)
+    library. The host name is of the format `name-of-event-hub-instance.servicebus.windows.net`.
 
-Please take a look at the [examples](https://github.com/Azure/azure-sdk-for-js/tree/master/sdk/eventhub/event-hubs/samples) directory for detailed examples.
+### Examples
 
-## Example 1 - Get the partition IDs.
+The following sections provide code snippets that cover some of the common tasks using Azure Event Hubs
 
-```js
-const { EventHubClient } = require('@azure/event-hubs');
+- [Get the partition Ids](#get-the-partition-ids)
+- [Send events](#send-events)
+- [Receive events](#receive-events)]
+- [Get an event hub runtime info](#get-an-event-hub-runtime-info)
+- [Get info about the specified partition](#get-info-about-the-specified-partition)
 
-const client = EventHubClient.createFromConnectionString(process.env["EVENTHUB_CONNECTION_STRING"], process.env["EVENTHUB_NAME"]);
+### Get the partition Ids
 
-async function main() {
-  const partitionIds = await client.getPartitionIds();
-}
+Once you have created an instance of an `EventHubClient` class, get the partition IDs
+using the [getPartitionIds](https://docs.microsoft.com/en-us/javascript/api/@azure/event-hubs/eventhubclient?view=azure-node-latest#getpartitionids--) function.
```

Comment:
A line here about why the user might need to do this can help. Is partitionId mandatory for sending and receiving events?

Reply:
No, partitionId is an optional parameter. This is a helper method. User can use this method to fetch the partition Ids and these partition Ids can be use to send/receive events.

Reply:
If it is a helper method then it doesn't need its own sample.
What partitionId is used if none is passed during send and receive operations?

Reply:
If partitionId  is not specified, EventHub will store the messages in a round-robin fashion among all the different partitions in the EventHub.

Do you still think we should remove this sample ?

Reply:
If we keep the sample, we should talk about why it is important. Talk about the below things
- In the absence of partitionId, how does send behave? Round robin vs partitionKey
- parititonId is mandatory for receive operations, so if user doesnt want to hard code the partitionId, they will have to use this to get the ids